### PR TITLE
WIP - Fix end of file error

### PR DESCRIPTION
**Pull request by orenbm**
In case the CHANGELOG file doesn't have a newline at its end, we get the error `NoMethodError: undefined method `split' for nil:NilClass` instead of something useful. 

We should fix this so that we raise a proper error.

### CODE REVIEW

**Comment by orenbm** [comment]
@jtuttle i spent some time on this and couldn't manage to get it done. Can you take a look when you have the time? You may have the knowledge that i don't.

I added a test that is failing now and should be passing once we fix this. Thanks!

**Comment by jtuttle** [comment]
@orenbm I spent the morning playing with this and it seems like you might have actually found a bug in the parser. I see no reason it should be returning a nil result without a failure reason. I've posted on the Google Group about it but considering the last post is from several years ago I'm not expecting much.

In any case, the commit I just made modifies the error message to include the line and column numbers when the failure reason is blank, so at least there will be something to go on when the newline is missing rather than a confusing error message about `split`.

**Comment by orenbm** [comment]
Nice @jtuttle ! Can you share the error that will be raised in case of a missing eof?

**Comment by orenbm** [comment]
Also - do you want to merge this or do you think we need another reviewer?

**Comment by jtuttle** [comment]
I'm not explicitly handling the end of file case, but it will raise a `ParseError` with a blank reason and the line and column numbers pointing to the very end of the file. This isn't great, but it's leagues better than an error originating from our error-handling code, which is what we had before. :)

Another set of eyes couldn't hurt, @izgeri any suggestions?

**Comment by sgnn7** [comment]
I've seen the `nil` problem on my work to support the rc versions. If I remember correctly, it would happen when it was unable to determine which rule to apply the data to. I'm guessing there's multiple ambiguous matching rules.

**Comment by izgeri** [comment]
it sounds like we've taken a step toward a better experience, and generally the changelogs across projects are set and change in predictable ways (ie it's uncommon for someone to get bit by an issue)

I'm ok with merging this in if @orenbm is in a good spot with his project, and we're a bit better in the project for the next time there's a parser error

incremental progress ftw

**Comment by orenbm** [comment]
thanks @izgeri :)

@jtuttle i added you as a reviewer, i think we can merge this.

**Review by jtuttle — APPROVED** [review]

**Pull request merged**
